src/System.Management.Automation/help/MUIFileSearcher.cs

```diff
@@ -133,13 +133,11 @@ private string[] GetFiles(string path, string pattern)
                     result.Add(filePath);
                     break;
                 }
-                // If the input is pattern instead of string, we need to use Regex expression.
-                if (pattern.Contains("*") || pattern.Contains("?"))
+
+                string fileName = Path.GetFileName(filePath);
```

Comment:
Actually, shouldn't `fileName` also be used when there are no wildcards? Currently
```PowerShell
if (filePath.IndexOf(pattern, StringComparison.OrdinalIgnoreCase) >= 0)
```
is searching the whole path for a match. If it is desirable to only target the file name when performing the wildcard match, shouldn't it also be appropriate here? (Actually, given that an exact match is expected — that's how `GetFiles` would work in the non-UNIX case for a non-wildcard pattern, if I am not mistaken — shouldn't the test be against `fileName.Equals(pattern, StringComparison.OrdinalIgnoreCase)` instead of using `IndexOf`?)

(An unrelated minor issue I believe exists with the logic of `GetFiles` is using `break` instead of `continue` when a non-wildcard match is found. This might limit the matches in the (rare) case in which a file name equals a wildcard search pattern, e.g. foo?, though I haven't tested how `GetFiles`would behave in that scenario, and it might be nitpicking if the expected directory and file names are curated.

Also, sorry for butting in, but since I wasn't sure enough about my observations I didn't want to open an issue, so I didn't know a better way to proceed.)